hw/block/nvme.c

```diff
@@ -2594,7 +2594,7 @@ static int lnvm_init_meta(LnvmCtrl *ln)
         ln->meta_auto_gen = 0;
     }
 
-    ln->metadata = fopen(ln->meta_fname, "w+");
+    ln->metadata = fopen(ln->meta_fname, "r+");
```

Comment:
What if the file doesn't exist?

Reply:
If you open with w+ then you overwrite the data, so there is no way to test recovery. Better not use the file if not created than to overwrite. Unless you have a better idea?

Reply:
If it doesn't exist, it should properly be created first. :)

Reply:
Hehehe. I'll fix that and resed tomorrow.